addons/interact_menu/functions/fnc_keyDown.sqf

```diff
@@ -37,27 +37,30 @@ GVAR(useCursorMenu) = (vehicle ACE_player != ACE_player) ||
                       {(_menuType == 1) && {(isWeaponDeployed ACE_player) || GVAR(AlwaysUseCursorSelfInteraction) || {cameraView == "GUNNER"}}} ||
                       {(_menuType == 0) && GVAR(AlwaysUseCursorInteraction)};
 
+// Delete existing controls in case there's any left
+GVAR(iconCount) = 0;
+for "_i" from 0 to (count GVAR(iconCtrls))-1 do {
+    ctrlDelete (GVAR(iconCtrls) select _i);
+    GVAR(ParsedTextCached) set [_i, ""];
+};
+GVAR(iconCtrls) resize GVAR(iconCount);
+
 if (GVAR(useCursorMenu)) then {
-    createDialog QGVAR(cursorMenu);
+    (findDisplay 46) createDisplay QGVAR(cursorMenu); //"RscCinemaBorder";//
+    (finddisplay 91919) displayAddEventHandler ["KeyUp", {[_this,'keyup'] call CBA_events_fnc_keyHandler}];
+    (finddisplay 91919) displayAddEventHandler ["KeyDown", {[_this,'keydown'] call CBA_events_fnc_keyHandler}];
     // The dialog sets:
     // uiNamespace getVariable QGVAR(dlgCursorMenu);
     // uiNamespace getVariable QGVAR(cursorMenuOpened);
-    ctrlEnable [91921, true];
     GVAR(cursorPos) = [0.5,0.5,0];
-    ((finddisplay 91919) displayctrl 91921) ctrlAddEventHandler ["MouseMoving", {
-        if (GVAR(cursorKeepCentered)) then {
-            GVAR(cursorPos) = GVAR(cursorPos) vectorAdd [_this select 1, _this select 2, 0] vectorDiff [0.5, 0.5, 0];
-            setMousePosition [0.5, 0.5];
-        } else {
-            GVAR(cursorPos) = [_this select 1, _this select 2, 0];
-        };
-    }];
-    // handles LMB in cursor mode when action on keyrelease is disabled
-    ((finddisplay 91919) displayctrl 91921) ctrlAddEventHandler ["MouseButtonDown", {
-        if !(GVAR(actionOnKeyRelease)) then {
-            [GVAR(openedMenuType),true] call FUNC(keyUp);
-        };
-    }];
+
+    _ctrl = (findDisplay 91919) ctrlCreate ["RscStructuredText", 9922];
+    _ctrl ctrlSetPosition [safeZoneX, safeZoneY, safeZoneW, safeZoneH];
+    _ctrl ctrlCommit 0;
```

Comment:
I had to switch to manually creating the control in which the mouse events are listened to. For some reason `ctrlAddEventHandler ["MouseMoving",` didn't work on static controls when using displays (used to work alright with dialogs)